### PR TITLE
Compound form auto mass

### PR DESCRIPTION
**Pull request by CWestICL**
The field for manual mass input has been removed from the `NewCompoundForm` form, and `dfndb.models.Compound.clean()` no longer checks for manual input. All help text and error messages referencing manual mass input have also been updated to reflect.

Closes https://github.com/ImperialCollegeLondon/Faraday-liionsden/issues/218

### CODE REVIEW

**Review by dandavies99 — APPROVED** [review]
This looks good to me @CWestICL! Is this PR still in draft or ready to merge?

File: dfndb/views.py
Comment (AdrianDAlessandro) [inline dfndb/views.py:63]: Not to be addressed in this PR, but I don't like that this change needed to be made twice. Perhaps there should be some unification between the "Create" and "Update" views.

**Review by AdrianDAlessandro — CHANGES_REQUESTED** [review]
This is great, there is one detail that still needs to be changed though. In the `__init__` method for the `NewCompoundForm` is still adds a `Column` object for the "mass" input. That should be removed and the "name" and "formula" columns re-sized appropriately.

File: dfndb/views.py
Comment (dandavies99) [reply dfndb/views.py:63]: Yes the challenge is that the `post()` method needs to be overridden for each view in a slightly different way (depending on whether the object exists already). There's probably a way to do this with some if statements.

File: dfndb/models.py
Comment (AdrianDAlessandro) [inline dfndb/models.py:33]: I wonder if this `mass` field should be converted into a property now? I noticed in the tests for this model that the mass is set on creation of the object in one of them (`TestCompound.test_unique_together`). We probably want to prevent that being possible now.

Maybe we remove the `clean` method entirely and replace it with a property for `mass`, @dandavies99 ?

**Review by AdrianDAlessandro — CHANGES_REQUESTED** [review]
This is great! You've done what I requested, but I've just thought of another thing. The tests for this model should now be updated. The relevant place to look is `tests/dfndb/test_models.py` in the `TestCompound` class. The pre-existing tests didn't actually check the `clean` method properly, otherwise they should've broken once you changed it.

File: dfndb/models.py
Comment (dandavies99) [reply dfndb/models.py:33]: Yes it's probably simpler just to move that logic into a property. There are a few other examples of where `@property` is used in that file for other models @CWestICL.

**Comment by AdrianDAlessandro** [comment]
@dandavies99 Can you check the changes I made before I approve it?

**Review by CWestICL — COMMENTED** [review]
Checked and tested, all looks good!

**Review by AdrianDAlessandro — APPROVED** [review]
I'm approving this, but want to wait for @dandavies99 to check my changes before it is merged.

**Review by dandavies99 — APPROVED** [review]
This behaves well now, nice to see specific error messages when an invalid formula is entered saying e.g. which character is unexpected.

**Pull request merged**
